### PR TITLE
Implement public poll participation features and fixes

### PR DESCRIPTION
**Pull request by OPS-PIvers**
This pull request introduces two main sets of changes: (1) it adds a new, fully tested public poll voting route and participant app, and (2) it fixes a subtle bug in the widget title editing logic that could cause cancelled edits to be incorrectly saved. Additionally, it enhances the admin configuration panel for the Need / Do / Put / Then widget, and adds an automated Claude code review workflow.

**New Features:**

* **Public Poll Voting App**
  - Adds a new lazy-loaded `PollVoteApp` React component and route (`/poll/:id`) for anonymous poll voting, including all necessary route handling and UI composition. The route is public and does not require teacher authentication. [[1]](diffhunk://#diff-a369d17a5dcc6a08663afe8ec644ed0ebc2624f39b704924078c01fd52c58fa3R104-R108) [[2]](diffhunk://#diff-a369d17a5dcc6a08663afe8ec644ed0ebc2624f39b704924078c01fd52c58fa3R413) [[3]](diffhunk://#diff-a369d17a5dcc6a08663afe8ec644ed0ebc2624f39b704924078c01fd52c58fa3R669-R684)
  - Comprehensive test suite for `PollVoteApp`, covering error states, voting, live tally updates, and edge cases (e.g., closed polls, re-vote failures).

**Bug Fixes:**

* **Widget Title Editing Regression**
  - Fixes a regression where pressing Escape while editing a widget title could still save the cancelled edit due to a stale closure in the `onBlur` handler. The fix uses a ref flag to prevent the Firestore write if editing is cancelled. Includes a regression test to ensure the bug does not recur. [[1]](diffhunk://#diff-0d33fd1f9e084a22794f65c762da0467ecb5d266454fa1359d74a6156b9c03dcR517-R532) [[2]](diffhunk://#diff-0d33fd1f9e084a22794f65c762da0467ecb5d266454fa1359d74a6156b9c03dcR2854-R2858) [[3]](diffhunk://#diff-a7455a8ac98e7a00d51e8f3e5c25eddac9794c773ad2cf6f0e48092d6a53f9deR1094-R1174)

**Admin Panel Improvements:**

* **Need / Do / Put / Then Widget Appearance Defaults**
  - Refactors and expands the admin configuration panel to allow setting building-level defaults for font family, text size, text color, card color, and card opacity for the Need / Do / Put / Then widget. This makes the widget’s initial appearance customizable per building. [[1]](diffhunk://#diff-af66d6dd618627e9a202b54920437706331acf75f68261e914824915f53f3140L2-R69) [[2]](diffhunk://#diff-af66d6dd618627e9a202b54920437706331acf75f68261e914824915f53f3140L39-R186)

**Developer Experience:**

* **Automated Claude Code Review**
  - Adds a GitHub Actions workflow (`claude-review.yml`) to automatically trigger high-signal Claude code reviews on PRs, excluding docs-only changes, and posting feedback directly to GitHub.

### CODE REVIEW

File: components/poll/PollVoteApp.tsx [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+                total === 0 ? 0 : Math.round((count / total) * 100);
+              const isMine = votedIndex === index;
+              return (
+                <button
+                  key={option.id}
+                  type="button"
+                  disabled={submitting}
+                  onClick={() => {
+                    void castVote(index);
+                  }}
```
[the remark this answers is not in the document]
Comment (OPS-PIvers) [reply components/poll/PollVoteApp.tsx:157]: Done in 144ec4f5 — option buttons are now disabled={submitting || closed}, so a closed poll no longer fires guaranteed-to-fail writes. Good catch.

File: components/poll/PollVoteApp.tsx [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+
+  // Subscribe to the live votes subcollection (read is open to any authed
+  // user — anonymous vote docs carry no PII). Sign in anonymously first so
+  // the read is authorized. This is synchronisation with an external system,
+  // which is what useEffect is for.
+  useEffect(() => {
+    if (!ready) return;
+    // eslint-disable-next-line @typescript-eslint/no-empty-function
+    let unsubscribe = () => {};
+    let cancelled = false;
```
[the remark this answers is not in the document]
Comment (OPS-PIvers) [reply components/poll/PollVoteApp.tsx]: Done in 144ec4f5 — removed the eslint-disable; the noop is now a typed `let unsubscribe: () => void = () => undefined;`, which avoids no-empty-function without a suppression comment.

File: components/widgets/PollWidget/Settings.tsx [hunk 3]
```diff
@@ -21,14 +23,43 @@ import {
...
+    } else {
+      void beginSession('fresh');
+    }
+  };
+
+  const handleStopClick = async () => {
+    if (!user) return;
+    const next = await stopPollSession(config, user.uid);
+    updateWidget(widget.id, { config: next });
+  };
```
[the remark this answers is not in the document]
Comment (OPS-PIvers) [reply components/widgets/PollWidget/Settings.tsx:74]: Done in 144ec4f5 — wrapped beginSession and handleStopClick in try/catch with console.error, matching the RemotePollControl pattern.

File: .github/workflows/claude-review.yml [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+      - name: Checkout repository
+        uses: actions/checkout@v6
+        with:
+          fetch-depth: 1
+
+      - name: Claude review
+        uses: anthropics/claude-code-action@v1
+        with:
+          # Pro/Max subscription auth â€” generate locally with `claude setup-token`
+          # and store as a repo secret. No metered API key required.
```
[the remark this answers is not in the document]
Comment (OPS-PIvers) [reply .github/workflows/claude-review.yml:41]: Done in 144ec4f5 — replaced the em-dash mojibake with ASCII.

File: .github/workflows/claude-review.yml [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+            - Adherence to existing patterns and conventions in the repo
+
+            Be concise and high-signal: only flag things that genuinely matter.
+            If the PR looks good, say so briefly rather than inventing nitpicks.
+
+            The PR branch is already checked out in the working directory.
+            Use `gh pr comment` for top-level feedback.
+            Use `mcp__github_inline_comment__create_inline_comment` (with
+            `confirmed: true`) for specific line-level issues.
+            Only post GitHub comments â€” do not return review text as a message.
```
[the remark this answers is not in the document]
Comment (OPS-PIvers) [reply .github/workflows/claude-review.yml:61]: Done in 144ec4f5 — replaced the em-dash mojibake with ASCII.

File: components/widgets/PollWidget/PollWidget.test.tsx [hunk 6]
```diff
@@ -47,6 +68,21 @@ describe('PollWidget', () => {
       updateWidget: mockUpdateWidget,
```
[the remark this answers is not in the document]
Comment (OPS-PIvers) [reply components/widgets/PollWidget/PollWidget.test.tsx:72]: Moved vi.clearAllMocks() to the top of beforeEach in 144ec4f5 for conventional ordering. One correction: clearAllMocks only resets call history (mockClear), not mockReturnValue/mockImplementation, so it was not actually wiping the stubs and the new tests were not order-dependent — but the top-of-block order reads clearer.

File: .github/workflows/claude-review.yml (hunk 4 above)
[the remark this answers is not in the document]
Comment (OPS-PIvers) [reply .github/workflows/claude-review.yml:41]: Update: reverted this in 82689766. The claude-code-action requires the PR-branch workflow file to be byte-identical to the default branch (else its OIDC token exchange 401s and the "Claude Code Review" check fails). main still has the em-dash here, so the ASCII cleanup has to land as a direct PR to main rather than via this feature branch.

File: .github/workflows/claude-review.yml (hunk 5 above)
[the remark this answers is not in the document]
Comment (OPS-PIvers) [reply .github/workflows/claude-review.yml:61]: Update: reverted this in 82689766. The claude-code-action requires the PR-branch workflow file to be byte-identical to the default branch (else its OIDC token exchange 401s and the "Claude Code Review" check fails). main still has the em-dash here, so the ASCII cleanup has to land as a direct PR to main rather than via this feature branch.

File: components/widgets/PollWidget/Widget.tsx [hunk 7]
```diff
@@ -46,7 +59,29 @@ export const PollWidget: React.FC<{ widget: WidgetData }> = ({ widget }) => {
...
+        setSessionTally(aggregateVotes(votes, options.length));
+      }
+    );
+    return () => {
+      unsub();
+      // Clear on teardown so a stopped session's counts don't flash on the
+      // board when a fresh session starts before its first snapshot arrives.
+      setSessionTally([]);
+    };
+  }, [activePollSessionId, user, options.length]);
```
[the remark this answers is not in the document]
Comment (OPS-PIvers) [reply components/widgets/PollWidget/Widget.tsx]: Addressed in 934fe09a — the votes effect now depends on the memoised `options` object instead of `options.length`, so exhaustive-deps is satisfied unambiguously (and since option editing is locked while a session is live, it does not re-subscribe spuriously). For the record, local eslint reported 0 warnings on the prior form too — `options` was already useMemo'd — so it was not actually failing, but this is the cleaner shape.
